<launch>
    <!-- =========================================================================================================== -->
    <!-- ===========================================NODE SETTINGS=================================================== -->
    <!-- =========================================================================================================== -->
    <!-- Global Settings For All Nodes -->
    <arg name="scenario_file" default="$(find rasberry_data_collection)/scripts/scenarios/manual_data_capture.yaml"/>

    <!-- Autonomous data collection -->
    <arg name="camera_sync_queue_size" default="10"/>
    <arg name="camera_sync_thresh" default="0.5"/>
    <arg name="camera_timeout" default="60"/>
    <arg name="camera_warmup" default="3"/>
    <arg name="save_location" default="data"/>
    <arg name="stabilise_time" default="1"/>

    <!-- Camera Settings -->
    <arg name="device_sync" default="false"/>
    <arg name="odom_frame_id" default="base_link"/>
    <arg name="initial_camera_reset" default="false"/>
    <arg name="rgb_max_side" default="1280"/>
    <arg name="rgb_min_side" default="720"/>
    <arg name="rgb_fps" default="6"/>
    <arg name="depth_max_side" default="848"/>
    <arg name="depth_min_side" default="480"/>
    <arg name="depth_fps" default="6"/>
    <arg name="align_depth_to_colour" default="true"/>
    <arg name="imu_enabled" default="false"/>
    <arg name="unite_imu_method" default="linear_interpolation"/>
    <arg name="infrared_enabled" default="true"/>
    <arg name="filters" default="disparity,spatial"/>

    <!-- =========================================================================================================== -->
    <!-- ===============================================NODES======================================================= -->
    <!-- Launch cameras -->
    <node pkg="rasberry_data_collection" type="dynamic_camera_launcher.py" name="rasberry_data_collection_cameras" output="screen">
        <param name="scenario_file" value="$(arg scenario_file)" type="str"/>

        <!-- Camera Parameters -->
        <param name="dyn_odom_frame_id" value="$(arg odom_frame_id)" type="str"/>
        <param name="dyn_filters" value="$(arg filters)" type="str"/>
        <param name="dyn_initial_reset" value="$(arg initial_camera_reset)" type="bool"/>
        <param name="dyn_device_sync" value="$(arg device_sync)" type="bool"/>
        <param name="dyn_rgb_max_side" value="$(arg rgb_max_side)" type="int"/>
        <param name="dyn_rgb_min_side" value="$(arg rgb_min_side)" type="int"/>
        <param name="dyn_rgb_fps" value="$(arg rgb_fps)" type="int"/>
        <param name="dyn_depth_max_side" value="$(arg depth_max_side)" type="int"/>
        <param name="dyn_depth_min_side" value="$(arg depth_min_side)" type="int"/>
        <param name="dyn_depth_fps" value="$(arg depth_fps)" type="int"/>
        <param name="dyn_align_depth_to_colour" value="$(arg align_depth_to_colour)" type="bool"/>
        <param name="dyn_imu_enabled" value="$(arg imu_enabled)" type="bool"/>
        <param name="dyn_unite_imu_method" value="$(arg unite_imu_method)" type="str"/>
        <param name="dyn_infrared_enabled" value="$(arg infrared_enabled)" type="bool"/>
    </node>

    <!-- Launch data collection scenario -->
    <node pkg="rasberry_data_collection" type="manual_data_capture.py" name="rasberry_data_collection_manual" output="screen">
        <param name="scenario_file" value="$(arg scenario_file)"/>
        <param name="rgb_enabled" value="True"/>
        <param name="depth_enabled" value="True"/>
        <param name="aligned_depth_enabled" value="$(arg align_depth_to_colour)"/>
        <param name="ir_enabled" value="$(arg infrared_enabled)"/>
        <param name="sync_queue_size" value="$(arg camera_sync_queue_size)"/>
        <param name="sync_thresh" value="$(arg camera_sync_thresh)"/>
        <param name="timeout" value="$(arg camera_timeout)"/>
        <param name="warmup" value="$(arg camera_warmup)"/>
        <param name="save_location" value="$(arg save_location)"/>
        <param name="stabilise_time" value="$(arg stabilise_time)"/>
    </node>
</launch>
